In [24]:
import pandas as pd
import numpy as np

# Описание данных:

* userID: уникальный идентификатор пользователя
* browser: браузер, который использовал userID
* slot: в каком статусе пользователь участвовал в исследовании (exp = видел измененную страницу, control = видел неизменную страницу)
* n_clicks: количество кликов, которые пользоваль совершил за n_queries
* n_queries: количество запросов, который совершил userID, пользуясь браузером browser
* n_nonclk_queries: количество запросов пользователя, в которых им не было совершено ни одного клика

Обращаем ваше внимание, что не все люди используют только один браузер, поэтому в столбце userID есть повторяющиеся идентификаторы. В предлагаемых данных уникальным является сочетание userID и browser.

In [25]:
data=pd.read_csv('ab_browser_test.csv')
data.head()

userID     browser     slot  n_clicks  n_queries  n_nonclk_queries
0       1  Browser #2      exp        23         32                19
1       3  Browser #4      exp         3          4                 2
2       5  Browser #4      exp        29         35                16
3       6  Browser #4  control        12          6                 0
4       7  Browser #4      exp        54         68                30

# Task 1

Основная метрика, на которой мы сосредоточимся в этой работе, — это количество пользовательских кликов на web-странице в зависимости от тестируемого изменения этой страницы.

Посчитайте, насколько в группе exp больше пользовательских кликов по сравнению с группой control в процентах от числа кликов в контрольной группе.

Полученный процент округлите до третьего знака после точки.

In [26]:
exp_clics=data[data['slot']=='exp']['n_clicks']
control_clics=data[data['slot']=='control']['n_clicks']
round ((exp_clics.sum()-control_clics.sum())/control_clics.sum()*100,3)

1.614

Task 1 answer=1.614

# Task 2

Давайте попробуем посмотреть более внимательно на разницу между двумя группами (control и exp) относительно количества пользовательских кликов.

Для этого постройте с помощью бутстрепа 95% доверительный интервал для средних значений и медиан количества кликов в каждой из двух групп. Отметьте все верные утверждения.

In [27]:
def get_bootstrap_samples(data, n_samples):
    indices = np.random.randint(0, len(data), (n_samples, len(data)))
    samples = data[indices]
    return samples
def stat_intervals(stat, alpha):
    boundaries = np.percentile(stat, [100 * alpha / 2., 100 * (1 - alpha / 2.)])
    return boundaries

In [ ]:
np.random.seed(0)
ilec_median_scores = list(map(np.median, get_bootstrap_samples(exp_clics, 1000)))